src/recipes.jl

```diff
@@ -274,10 +274,19 @@ function Makie.plot!(gp::GraphPlot)
             if $(gp.elabels_rotation) isa Real
                 # fix rotation to a single angle
                 rot = $(gp.elabels_rotation)
+            elseif $(gp.elabels_rotation) == automatic
+                #point the labels up
+                rot = broadcast($to_angle, edge_paths[], $positions, gp.elabels_shift[])
```

Comment:
I think that would be a good direction towards a more finely grained interface. I specially like the `Dict` for which then the user doesn't need to handle in a whole matrix rather only the edges/vertices of interest. Do you think the user could also supply a `Dict` of type `Dict{Edge, Any}` or strictly `Dict{Integer, Any}` ?

Reply:
Interesting, I havn't thought about the `Dict{Edge,Any}` option. My gut feeling is that this is probably something to think about in a separate PR. I like it, because the importance of the edge iterator order feels weird. However, the dependence on the order of the edge iterator is basically baked into every part of the recipe...

Reply:
> However, the dependence on the order of the edge iterator is basically baked into every part of the recipe

Yes, this is why I got excited with that `Dict` compatibility of the `getattr` function. Maybe using this function one can turn it around ?
And for the cases where such order is very much hardwired, maybe a wrapper can be built to translate data structures e.g. from `(Edge(6 => 10) => :blue )` to `[:automatic, :automatic, ..., :automatic, :blue, :automatic, ...]`

But yeah, probably a different issue/PR about this would make sense.